homework-2/src/js/views/viewRestorePassword.js

```diff
@@ -0,0 +1,77 @@
+class ViewRestorePassword {
+  constructor() {
+    this.app = this.getElement('#root');
+    this.view = '/restorePassword';
+  }
+
+  getElement(selector) {
+    const element = document.querySelector(selector);
+    return element;
+  }
+
+  getElements(selector) {
+    const element = document.querySelectorAll(selector);
+    return element;
+  }
+
+  _initLocalListeners(addUser, onNavClick) {
+    this.getElement('.form-main').addEventListener('submit', (e) => {
+      const boolArr = [];
+      e.preventDefault();
+
+      e.path[0].querySelectorAll('input').forEach((item) => {
+        if (item.type == 'email' || item.type == 'password') {
+          let label = document.querySelector(`[for="${item.id}"]`);
+          const value = item.value;
+          const regul = new RegExp(
+            item.getAttribute('type') == 'email'
+              ? /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+              : /(?=.*[A-Z]+)(?=.*[0-9])(?=.*[a-z]){6,}/,
+          );
+
+          if (!regul.test(value)) {
+            item.classList.add('is-error-input');
+            label.classList.replace('is-hidden-error', 'is-visible-error');
+          } else {
+            item.classList.remove('is-error-input');
+            label.classList.replace('animate__backInUp', 'animate__backOutDown');
+
+            setTimeout(() => {
+              label.classList.replace('is-visible-error', 'is-hidden-error');
+              label.classList.replace('animate__backOutDown', 'animate__backInUp');
+            }, 1000);
+          }
+        }
+      });
```

Comment:
Alot of code duplicates in different files, create additional util files to prevent duplicates.

Reply:
Removed code duplication. Made inheritance from the main class